pkg/resources/user_password_policy_attachment.go

```diff
@@ -0,0 +1,169 @@
+	},
+}
+
+func UserPasswordPolicyAttachment() *schema.Resource {
+	return &schema.Resource{
+		Description: "Specifies the password policy to use for a certain user.",
+
+		Create: CreateUserPasswordPolicyAttachment,
+		Read:   ReadUserPasswordPolicyAttachment,
+		Delete: DeleteUserPasswordPolicyAttachment,
+
+		Schema: userPasswordPolicyAttachmentSchema,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: func(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+				parts := strings.Split(d.Id(), helpers.IDDelimiter)
+				if len(parts) != 4 {
+					return nil, fmt.Errorf("id should be in the format 'database|schema|password_policy|user_name|roles', but I got '%s'", d.Id())
+				}
+				passwordPolicyDatabase := sdk.NewAccountIdentifierFromFullyQualifiedName(parts[0])
+				passwordPolicySchema := sdk.NewAccountIdentifierFromFullyQualifiedName(parts[1])
+				passwordPolicyName := sdk.NewAccountIdentifierFromFullyQualifiedName(parts[2])
+				userName := sdk.NewAccountObjectIdentifierFromFullyQualifiedName(parts[3])
+				if err := d.Set("password_policy_database", passwordPolicyDatabase.Name()); err != nil {
+					return nil, err
+				}
+				if err := d.Set("password_policy_schema", passwordPolicySchema.Name()); err != nil {
+					return nil, err
+				}
+				if err := d.Set("password_policy_name", passwordPolicyName.Name()); err != nil {
+					return nil, err
+				}
+				if err := d.Set("user_name", userName.Name()); err != nil {
+					return nil, err
+				}
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+	}
+}
+
+func CreateUserPasswordPolicyAttachment(d *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	client := sdk.NewClientFromDB(db)
+	ctx := context.Background()
+
+	userName := sdk.NewAccountObjectIdentifierFromFullyQualifiedName(d.Get("user_name").(string))
+	passwordPolicyDatabase := sdk.NewAccountIdentifierFromFullyQualifiedName(d.Get("password_policy_database").(string))
+	passwordPolicySchema := sdk.NewAccountIdentifierFromFullyQualifiedName(d.Get("password_policy_schema").(string))
+	passwordPolicyName := sdk.NewAccountIdentifierFromFullyQualifiedName(d.Get("password_policy_name").(string))
+	passwordPolicy := sdk.NewSchemaObjectIdentifier(passwordPolicyDatabase.Name(), passwordPolicySchema.Name(), passwordPolicyName.Name())
+
+	err := client.Users.Alter(ctx, userName, &sdk.AlterUserOptions{
+		Set: &sdk.UserSet{
+			PasswordPolicy: passwordPolicy,
+		},
+	})
+	if err != nil {
+		return err
+	}
+	if err := d.Set("password_policy_database", passwordPolicyDatabase.Name()); err != nil {
+		return err
+	}
+	if err := d.Set("password_policy_schema", passwordPolicySchema.Name()); err != nil {
+		return err
+	}
+	if err := d.Set("password_policy_name", passwordPolicyName.Name()); err != nil {
+		return err
+	}
+	if err := d.Set("user_name", helpers.EncodeSnowflakeID(userName)); err != nil {
+		return err
+	}
```

Comment:
You can remove all of the `d.Set` functions. They should be called in the Read function that is called as the last step of create, here: `return ReadUserPasswordPolicyAttachment(d, meta)`